packages/amplication-client/src/VersionControl/Commit.tsx

```diff
@@ -156,47 +190,55 @@ const Commit = ({ projectId, noChanges }: Props) => {
               {!loading && (
                 <GlobalHotKeys keyMap={keyMap} handlers={handlers} />
               )}
-              <TextField
-                rows={3}
-                textarea
-                name="message"
-                label={noChanges ? "Build message" : "Commit message..."}
-                disabled={loading}
-                autoFocus
-                hideLabel
-                placeholder={noChanges ? "Build message" : "Commit message..."}
-                autoComplete="off"
-              />
-
+              {showCommitMessage && (
+                <TextField
+                  rows={3}
+                  textarea
+                  name="message"
+                  label={noChanges ? "Build message" : "Commit message..."}
+                  disabled={loading}
+                  autoFocus
+                  hideLabel
+                  placeholder={
+                    noChanges ? "Build message" : "Commit message..."
+                  }
+                  autoComplete="off"
+                />
+              )}
               <LicenseIndicatorContainer
                 featureId={BillingFeature.BlockBuild}
                 licensedResourceType={LicensedResourceType.Project}
                 licensedTooltipText="The workspace reached your plan's project limitation. "
               >
-                <Button
-                  type="submit"
-                  buttonStyle={EnumButtonStyle.Primary}
-                  eventData={{
-                    eventName: AnalyticsEventNames.CommitClicked,
-                  }}
-                  disabled={loading}
-                >
-                  {noChanges ? "Rebuild" : "Commit changes & build "}
-                </Button>
+                {commitBtnType === CommitBtnType.Button ? (
+                  <Button
+                    type="submit"
+                    buttonStyle={EnumButtonStyle.Primary}
+                    eventData={{
+                      eventName: AnalyticsEventNames.CommitClicked,
+                    }}
+                    disabled={loading}
+                  >
+                    {noChanges ? "Rebuild" : "Commit changes & build "}
+                  </Button>
+                ) : commitBtnType === CommitBtnType.JumboButton ? (
+                  <JumboButton
+                    text="Generate the code for my new architecture"
```

Comment:
you may need to wrap the JumboButton in another component to get the form from the FormikContext and use form.submit on click